dask/tests/test_config.py

```diff
@@ -489,3 +491,26 @@ def test_get_override_with():
             "hello": "world"
         }
         assert dask.config.get("foo", override_with=["one"]) == ["one"]
+
+
+def test_config_serialization():
+    # Use context manager without changing the value to ensure test side effects are restored
+    with dask.config.set({"array.svg.size": dask.config.get("array.svg.size")}):
+
+        # Take a round trip through the serialization
+        serialized = serialize({"array": {"svg": {"size": 150}}})
+        config = deserialize(serialized)
+
+        dask.config.update(dask.config.global_config, config)
+        assert dask.config.get("array.svg.size") == 150
+
+
+def test_config_inheritance():
+    os.environ["DASK_INTERNAL_INHERIT_CONFIG"] = serialize(
```

Comment:
You could instead/also test `collect_env` by passing a dictionary (I notice that the env parameter is not documented there).

I don't insist, happy for merge as is.

Reply:
Thanks @martindurant that makes the test much neater.